src/coreclr/jit/codegencommon.cpp

```diff
@@ -7839,6 +7850,30 @@ void CodeGen::genReturn(GenTree* treeNode)
 #endif // defined(DEBUG) && defined(TARGET_XARCH)
 }
 
+#ifdef SWIFT_SUPPORT
+//------------------------------------------------------------------------
+// genSwiftErrorReturn: Generates code for storing error value in REG_SWIFT_ERROR.
+//
+// Arguments:
+//    treeNode - The GT_SWIFT_ERROR_RET tree node,
+//    where its operand is the SwiftError::Value field
+//
+// Return Value:
+//    None
+//
+void CodeGen::genSwiftErrorReturn(GenTreeUnOp* treeNode)
+{
+    assert(compiler->info.compCallConv == CorInfoCallConvExtension::Swift);
+    assert(compiler->lvaSwiftErrorArg != BAD_VAR_NUM);
+    assert(compiler->lvaSwiftErrorLocal != BAD_VAR_NUM);
+
+    GenTree* op = treeNode->gtGetOp1();
+
+    const regNumber reg = genConsumeReg(op);
+    inst_Mov(op->TypeGet(), REG_SWIFT_ERROR, reg, true, EA_PTRSIZE);
```

Comment:
I don't think anything prevents LSRA from allocating `REG_SWIFT_ERROR` to something that is used after this node (e.g. by the actual `GT_RETURN`).

I think the most natural way to represent this is to add a new variant of `GT_RETURN` that is binary and has both the actual return value and the swift error as operands, and that we can create during import.

Reply:
I can change the representation to that. Are there any situations where we currently might return a value in a callee-save register?

Reply:
I don't think so. The current problem is that we might have enregistered locals in the swift register that the `GT_RETURN` after this one uses, and then those values will be corrupted. I would expect it to be easy to hit the problem with some more tests that have both error returns and normal returns.

Reply:
It may not be the most natural shape, but could we ensure that `GT_SWIFT_ERROR_RET` is the last node in the `BBJ_RETURN` block by creating it while lowering `GT_RETURN`, and inserting it at the end of the block? We do something similar when creating P/Invoke epilog IR. I'm trying this locally, and it seems to work fine, plus we don't have to handle `GT_SWIFT_ERROR_RET` in any of the frontend phases.

Reply:
The problem with creating this node late in the JIT is that it does not naturally represent that there is a use of the swift error local at the point of all the `GT_RETURN` nodes. This means that we either permanently decide to treat it conservatively (what you did by marking it as address exposed) or we need to special case that all `GT_RETURN` nodes have an additional use of the swift error local everywhere that reasons about uses (not really feasible I think).

I think what you suggest works, but it locks us into never handling the error efficiently. We will be spilling and restoring it from stack even in optimized codegen and we won't be able to reason about its value.

Did you find it to be a lot of work to do it the natural way? Normally adding new nodes is not that bad, but since `GT_RETURN` has some special status it might be harder.

Reply:
I'm ok with just address exposing it always for now. We can do the work in the future if we really have appetite.

I think in this case you do not need `GT_SWIFT_ERROR_RET` at all. You can simply update the `GT_RETURN` codegen to also load the swift error register from the error local when it exists. For good measure I think you should also mark the error register as implicitly referenced.

Reply:
> Did you find it to be a lot of work to do it the natural way? Normally adding new nodes is not that bad, but since GT_RETURN has some special status it might be harder.

I think most of the awkwardness comes from `GT_SWIFT_ERROR_RET` trying to model `GT_RETURN` in most places, but the former is a binary op, and the latter is a unary op. Quirking around these situations is trivial, but it doesn't feel elegant.

> I'm ok with just address exposing it always for now. We can do the work in the future if we really have appetite.

Ok, I'll leave `GT_SWIFT_ERROR_RET` out of this PR for now, and start a follow-up PR for it.